chapter07/chapter07.adoc

```diff
@@ -0,0 +1,248 @@
+
+=== Metric Registry
+
+The *MetricRegistry* component acts as a container for storing and managing metrics within an application. It provides a structured way to collect, organize, and access various types of metrics (e.g., counters, gauges, histograms, and timers) for monitoring the behavior and performance of applications. It offers a centralized repository where metrics can be created and retrieved. This allows applications to consistently monitor critical operational and performance statistics.
+
+==== Types of Metric Registries
+
+MicroProfile Metrics creates metric registries for each scope:
+
+- Application Scope (`MetricRegistry.Type.APPLICATION`): Contains custom metrics that are specific to the application. These are typically the metrics that developers explicitly create and register to monitor application-specific behaviors.
+
+- Base Scope (`MetricRegistry.Type.BASE`): Contains metrics that are fundamental and common across all MicroProfile applications. These metrics provide basic information about the underlying JVM and application server.
+
+- Vendor Scope (`MetricRegistry.Type.VENDOR`): Contains metrics that are specific to the implementation of the MicroProfile platform being used. These metrics offer insights into vendor-specific features and optimizations.
+
+A metric registry is created as per the above scopes to enable the organization of metrics based on their origin and relevance.
+
+==== Instrumenting Microservices with MicroProfile Metrics
+
+Instrumenting microservices with MicroProfile Metrics enables developers to gain detailed insights into their application's operational health and performance. This level of observability is essential for maintaining scalable and resilient microservice architectures in dynamic environments.
+
+==== Tracking response time using `@Timed`
+
+MicroProfile Metrics also allows you to track a method's response time as a timed metric. The code example below shows how to use the @Timed annotation to track the response time.
+
+[source]
+----
+import org.eclipse.microprofile.metrics.annotation.Timed;
+// …
+
+public class ProductResource {
+
+    // …
+    // Expose the response time as a timer metric
+    @Timed(name = "productLookupTime",
+            tags = {"method=getProduct"},
+            absolute = true, 
+            description = "Time spent looking up products")
+    public Product getProduct(@PathParam("id") Long productId) {
+        return productService.getProduct(productId);
+    }
+
+    // …
+----
+  
+It will expose a metric called `productLookupTime`, which will track the amount of time spent in the `getProduct()` method in seconds.
+You can visit the following URL `++https://localhost:<port>/metrics?scope=application++` (Replace `<port>` with the actual port where the server is running) to see the response time of this method as below:
+
+[source]
+----
+…
+# HELP productLookupTime_seconds_max Time spent to lookup up products
```

Comment:
Time spent looking up products

Reply:
Fixed.

Reply:
@ttelang , you updated to "Time spent to looking up products", which is still incorrect.